cli/ops/io.rs

```diff
@@ -323,21 +339,41 @@ where
       panic!("poll a Read after it's done");
     }
 
-    let mut table = inner.state.lock_resource_table();
-    let resource = table
-      .get_mut::<StreamResource>(inner.rid)
-      .ok_or_else(bad_resource)?;
-    let nwritten = match DenoAsyncWrite::poll_write(
-      Pin::new(resource),
-      cx,
-      inner.buf.as_ref(),
-    ) {
-      Poll::Ready(Ok(v)) => v,
-      Poll::Ready(Err(err)) => return Poll::Ready(Err(err)),
-      Poll::Pending => return Poll::Pending,
-    };
-    inner.io_state = IoState::Done;
-    Poll::Ready(Ok(nwritten as i32))
+    if inner.io_state == IoState::Pending {
+      let mut table = inner.state.lock_resource_table();
+      let resource = table
+        .get_mut::<StreamResource>(inner.rid)
+        .ok_or_else(bad_resource)?;
+
+      let nwritten = match DenoAsyncWrite::poll_write(
+        Pin::new(resource),
+        cx,
+        inner.buf.as_ref(),
+      ) {
+        Poll::Ready(Ok(v)) => v,
+        Poll::Ready(Err(err)) => return Poll::Ready(Err(err)),
+        Poll::Pending => return Poll::Pending,
+      };
+      inner.io_state = IoState::Flush;
+      inner.nwritten = nwritten as i32;
+    }
```

Comment:
@ry regarding our conversation: after I added implicit flush in write tests started to pass, but two of them from `std`.
Those tests are: `testPrettierWithSpecifiedIgnore` and `testPrettierWithAutoIgnore`. Both are hanging during output to stdout on write 😡 it happens on second call `await stdout.write()` and hangs during `poll_write`. As soon as I change it to:
```rust
eprintln!("before write");
let nwritten = match DenoAsyncWrite::poll_write(
  Pin::new(resource),
  cx,
  inner.buf.as_ref(),
) {
  Poll::Ready(Ok(v)) => v,
  Poll::Ready(Err(err)) => return Poll::Ready(Err(err)),
  Poll::Pending => return Poll::Pending,
};
eprintln!("after write");
inner.io_state = IoState::Flush;
inner.nwritten = nwritten as i32;
```
(so adding just two prints) it starts working... 🙄 what the?

Reply:
From [Tokio docs](https://docs.rs/tokio/0.2.6/tokio/io/struct.Stdout.html):
```
The handle implements the AsyncWrite trait, but beware that concurrent writes to Stdout must be executed with care.
```

We use raw file handle but I guess the same idea applies. I removed concurrent calls to stdout.write, let's see how that works.